### PR TITLE
Adding tests from openssl

### PR DESCRIPTION
**Pull request by nasastry**
Adding tests from openssl, to run unit tests cases

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
JOB ID     : 7db997b70018a053db0e2df5d8ff89783e29376b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-09-29T05.16-7db997b/job.log
 (1/1) openssl-tests.py:OpenSSL.test:  FAIL: 10 test(s) failed, please refer to the log (734.65 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-09-29T05.16-7db997b/results.html
JOB TIME   : 735.44 s
```

File: security/openssl-tests.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        '''
+        Install the basic packages to support openssl
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        deps = ['gcc', 'make']
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        url = "https://github.com/openssl/openssl"
```
Comment (PraveenPenguin) [inline security/openssl-tests.py:38]: can we have zip file as url using that way it help to use asset fetch and lib provide cache dir and others useful param that avoid download very7 frequently

**Comment by nasastry** [comment]
RHEL:
```
Fetching asset from openssl-tests.py:OpenSSL.test
JOB ID     : 20ca561c9099fbaac541a34d4f8d1e4f778f03cc
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-02T13.20-20ca561/job.log
 (1/1) openssl-tests.py:OpenSSL.test: FAIL: 10 test(s) failed, please refer to the log (413.18 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-02T13.20-20ca561/results.html
JOB TIME   : 414.15 s
```

SLES:
```

Fetching asset from openssl-tests.py:OpenSSL.test
JOB ID     : 0fbb217a464ad1c48c22f6132142e4d36ae99b35
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-06T09.32-0fbb217/job.log
 (1/1) openssl-tests.py:OpenSSL.test: FAIL: 10 test(s) failed, please refer to the log (537.29 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-06T09.32-0fbb217/results.html
JOB TIME   : 539.33 s
```

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Pull request merged**